### PR TITLE
Implement Twisted Edwards Affine Coordinates

### PR DESCRIPTION
**Pull request by CPerezz**
This PR pretends to close: #50 and #66. 

Implementations done:
- Affine Coordinates definition & implementation.
- Point conversions between all Tw Eds coordinates defined on the library.
- Fixed point doubling and Scalar mul issues.
- Implemented tests for point ops.
- Implemented Constant Time equality verification between all of the implemented coordinate types relying on Affine Coordinates.
- Update docs.

Now all of the point `Eq` implementations rely over on `ConstantTimeEq` trait implementation done for `AffineCoordinates`.

### CODE REVIEW

File: src/backend/u64/field.rs [hunk 1]
```diff
@@ -2,6 +2,10 @@
 //! using 64-bit limbs with 128-bit products.
 //! In the 64-bit backend implementation, a `FieldElement` is
 //! represented in radix `2^52`.
+//! 
+//! The basic modular operations have been taken from the 
+//! [Curve25519-dalek repository](https://github.com/dalek-cryptography/curve25519-dalek) and refactored to work 
+//! for the Doppio finite field.
```
Comment (Bounce23) [inline src/backend/u64/field.rs]: This can also be sourced from other places so lets speak about citation.

File: src/edwards.rs [hunk 2]
```diff
@@ -247,6 +247,27 @@ impl<'a> From<&'a ProjectivePoint> for EdwardsPoint {
...
+    /// In affine form, each elliptic curve point has 2 coordinates, 
+    /// like (x,y). In the new projective form, 
+    /// each point will have 3 coordinates, like (X,Y,Z), 
+    /// with the restriction that Z is never zero.
+    ///  
+    /// The forward mapping is given by (x,y)→(xz,yz,z), 
+    /// for any non-zero z (usually chosen to be 1 for convenience).
+    /// 
+    /// After this is done, we move from Projective to Extended by
+    /// setting the new coordinate `T = X * Y`.
```
Comment (Bounce23) [inline src/edwards.rs]: Perhaps having an indication of how to find the examples would be useful.

File: src/backend/u64/field.rs (hunk 1 above)
Comment (CPerezz) [reply src/backend/u64/field.rs]: I cited the library from where the ops were taken an then refactored and the authors. What else should we add?

File: src/edwards.rs (hunk 2 above)
Comment (CPerezz) [reply src/edwards.rs]: What do you purpose? I think it's explained there correctly. What can I add?

**Comment by CPerezz** [comment]
Codecov/patch got a diff reduction, but the tests are built. So we merge the PR anyway.

**Pull request merged**